sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/util/common/CounterSet.java

```diff
@@ -125,16 +125,16 @@ public synchronized boolean add(Counter<?> e) {
     if (null == e) {
       return false;
     }
-    if (counters.containsKey(e.getName())) {
+    if (counters.containsKey(e.getFlatName())) {
```

Comment:
Given that e supports equality and hashCode, we shouldn't need to use `e.getFlatName()` as the key in the map -- just use the counter.